File: spec/php/RepeatNStrzTest.php
<?php
namespace Kaitai\Struct\Tests;

class RepeatNStrzTest extends TestCase {
    public function testRepeatNStrz() {
        $r = RepeatNStrz::fromFile(self::SRC_DIR_PATH . "/repeat_n_strz.bin");

        $this->assertEquals(2, $r->qty(), 2);
        $this->assertEquals(["foo", "bar"], $r->lines());
    }
}

File: spec/go/repeat_n_strz_test.go
package spec

import (
	"os"
	"testing"

	"github.com/kaitai-io/kaitai_struct_go_runtime/kaitai"
	"github.com/stretchr/testify/assert"

	. "test_formats"
)

func TestRepeatNStrz(t *testing.T) {
	f, err := os.Open("../../src/repeat_n_strz.bin")
	if err != nil {
		t.Fatal(err)
	}
	s := kaitai.NewStream(f)

	var r RepeatNStrz
	err = r.Read(s, &r, &r)
	if err != nil {
		t.Fatal(err)
	}

	assert.EqualValues(t, 2, r.Qty)
	assert.EqualValues(t, []string{"foo", "bar"}, r.Lines)
}
